### DISCUSSION TITLE
No api key found in request

### DISCUSSION THREAD

**Discussion by Asumalsingh** [supabase/supabase#7761] [Questions]
when I use authentication with next-js then we get an error "No API key found in request". Can anyone help me to resolve this error

**Comment by GaryAustin1** [comment]
What call are you making?
Is your apikey header set?

**Reply by orisa** [reply]
I'm currently facing this issue while trying to call the /logout endpoint using the rest API. It was working 2 days ago. I have no clue what could have changed.

**Comment by wcooperscott** [comment]
Please read the [NextJS Quickstart Guide](https://supabase.com/docs/guides/with-nextjs#initialize-a-nextjs-app). 

Pay special attention to section [Getting the API Key](https://supabase.com/docs/guides/with-nextjs#get-the-api-keys) and [Initialize a NextJS Project](https://supabase.com/docs/guides/with-nextjs#initialize-a-nextjs-app). 

It sounds like your environment is not setup correctly and you are not passing the API key and correct project URL when initializing your Supabase client.

```
const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL
const supabaseAnonKey = process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY

export const supabase = createClient(supabaseUrl, supabaseAnonKey)
```

**Reply by orisa** [reply]
I had to update my axios request. Thanks

**Reply by wcooperscott** [reply]
Glad you found a solution however you should not be connecting to Supabase with Axios. Your life will be 10x easier if you setup the SupabaseJS client for connecting to your database.

Please read [Initialize a NextJS app](https://supabase.com/docs/guides/with-nextjs#initialize-a-nextjs-app) and follow the instructions.

**Comment by Arisz7** [comment]
Hello im gettin the same error but in my case is a project that i accidently delete from vercel and it was working just fine, i did not change anything from my github repositoy. Im getting this error "message":"No API key found in request","hint":"No `apikey` request header or url param was found. with a 406 status does anyone know what this could be ? even tho i have set up the keys correctly cause i already checked for them in the headers of the request and they are the same as the one that i have set up in supabase :(

**Reply by Arisz7** [reply]
and in my project as well i have an .env.local in which one i provide all the keys from supabase

**Reply by vkornushenko** [reply]
have you set the environment variables  in vercel after create a new project?
You need to set them in vercel -> your project -> settings -> environment variables
And then set there variables from yours project **.env.local** file
